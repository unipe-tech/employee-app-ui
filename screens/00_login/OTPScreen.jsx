import { Icon } from "@react-native-material/core";
import { useNavigation } from "@react-navigation/core";
import { useEffect, useState } from "react";
import { Alert, BackHandler, SafeAreaView, Text, View } from "react-native";
import CountDown from "react-native-countdown-component";
import { useDispatch, useSelector } from "react-redux";
import Otp from "../../apis/login/Otp";
import Verify from "../../apis/login/Verify";
import SVGImg from "../../assets/UnipeLogo.svg";
import FormInput from "../../components/atoms/FormInput";
import Header from "../../components/atoms/Header";
import { COLORS, SIZES } from "../../constants/Theme";
import { KeyboardAvoidingWrapper } from "../../KeyboardAvoidingWrapper";
import { addCurrentScreen } from "../../store/slices/navigationSlice";
import { setLoginTimer } from "../../store/slices/timerSlice";
import { styles } from "../../styles";

const OTPScreen = () => {
  const dispatch = useDispatch();
  const navigation = useNavigation();

  const [otp, setOtp] = useState("");
  const [disabled, setDisabled] = useState(true);
  const [back, setBack] = useState(false);

  const countDownTime = useSelector((state) => state.timer.login);
  const phoneNumber = useSelector((state) => state.auth.phoneNumber);

  useEffect(() => {
    dispatch(addCurrentScreen("Otp"));
  }, []);

  useEffect(() => {
    if (otp.length === 6) {
      setDisabled(false);
    } else {
      setDisabled(true);
    }
  }, [otp]);

  const backAction = () => {
    if (!back) {
      Alert.alert("OTP Timer", "You must wait for 2 minutes to resend OTP.");
    } else {
      Alert.alert("Hold on!", "Do you want to update your phone number ?", [
        { text: "No", onPress: () => null, style: "cancel" },
        { text: "Yes", onPress: () => navigation.navigate("Login") },
      ]);
    }
    return true;
  };

  useEffect(() => {
    BackHandler.addEventListener("hardwareBackPress", backAction);
    return () =>
      BackHandler.removeEventListener("hardwareBackPress", backAction);
  }, []);

  return (
    <SafeAreaView style={styles.safeContainer}>
      <Header title="OTP" onLeftIconPress={() => backAction()} />
      <KeyboardAvoidingWrapper>
        <View style={styles.container}>
          <SVGImg style={styles.logo} />
          <Text style={styles.headline}>
            {" "}
            Please wait, we will auto verify the OTP {"\n"} sent to{" "}
            {phoneNumber}
            {back ? (
              <Icon
                name="edit"
                size={12}
                color={COLORS.primary}
                onPress={() => navigation.navigate("Login")}
              />
            ) : (
              <Icon
                name="edit"
                size={12}
                color={COLORS.gray}
                onPress={() =>
                  Alert.alert(
                    "OTP Timer",
                    "You must wait for 2 minutes to edit number."
                  )
                }
              />
            )}
          </Text>
          <FormInput
            containerStyle={{
              marginTop: 30,
              width: SIZES.width * 0.6,
              alignSelf: "center",
            }}
            letterSpacing={SIZES.width * 0.0699}
            autoFocus={true}
            value={otp}
            onChange={setOtp}
            maxLength={6}
            keyboardType="numeric"
            placeholder={"******"}
          />

          <CountDown
            until={countDownTime}
            onFinish={() => {
              setBack(true);
            }}
            size={20}
            style={{ marginTop: 20 }}
            digitStyle={{ backgroundColor: "#FFF" }}
            digitTxtStyle={{ color: COLORS.primary }}
            timeToShow={["M", "S"]}
            timeLabels={{ m: "MM", s: "SS" }}
            onChange={(time) => {
              dispatch(setLoginTimer(time));
            }}
          />
          {back ? <Otp /> : null}
          <Text style={styles.otpreadtxt}>
            {" "}
            Sit back & relax while we fetch the OTP & log you inside the Unipe
            App
          </Text>
          <Verify disabled={disabled} otp={otp}/>
        </View>
      </KeyboardAvoidingWrapper>
    </SafeAreaView>
  );
};

export default OTPScreen;
