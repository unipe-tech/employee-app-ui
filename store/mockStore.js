const store = {
  // Auth

  auth: {
    id: 1,
    phoneNumber: "9876543210",
    verifyStatus: "PENDING",
  },

  // Aadhaar

  aadhaar: {
    data: {
      aadhaar_data: {
        name: "Xyz garg",
      },
    },
    number: "789456123010",
    submitOTPtxnId: "987",
    verifyMsg: "This otp has been sent",
    verifyStatus: "PENDING",
  },

  // Bank

  bank: {
    accountNumber: "157894561230",
    ifsc: "INDB0000123",
    upi: "9876543210@upi",
    accountHolderName: "Xyz Goyal",
    verifyStatus: "PENDING",
    verifyMsg: "The verification is pending",
    bankName: "IndusInd Bank",
    bankBranch: "Mumbai",
    branchCity: "Mumbai",
  },

  // ESIC

  esic: {
    portal: { estCode: "", ipNumber: "" },
    familyDetails: {
      fatherHusband: { relation: "", name: "" },
      nominee: { relation: "", name: "" },
    },
    address: {
      present: { street: "", state: "", district: "", pincode: "" },
      permanent: { street: "", state: "", district: "", pincode: "" },
      nominee: { street: "", state: "", district: "", pincode: "" },
    },
  },

  // License

  license: {
    number: "",
    name: "",
    dob: "",
    bloodGroup: "",
    classes: {},
    rto: {},
    validity: {},
    verifyMsg: "",
    verifyStatus: "PENDING",
  },

  // Navigation

  navigation: { currentScreen: "Welcome" },

  // PAN

  pan: {
    dob: "02/02/2002",
    email: "xyz@unipe.co",
    gender: "Male",
    name: "Xyz Goyal",
    number: "ABCDE1234F",
    verifyMsg: "Verification Pending",
    verifyStatus: "PENDING",
  },

  // Profile

  profile: {
    maritalStatus: "Married",
    educationalQualification: "12th pass",
    alternatePhone: "+913216549870",
    email: "abc@unipe.co",
    selfie: "",
  },

  // Timer

  timer: { login: 2 * 60, aadhaar: 10 * 60 },
};

export default store;
