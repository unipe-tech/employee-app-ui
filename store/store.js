import { configureStore } from "@reduxjs/toolkit";
import authSlice from "./slices/authSlice";
import aadhaarSlice from "./slices/aadhaarSlice";
import panSlice from "./slices/panSlice";
import bankSlice from "./slices/bankSlice";
import esicSlice from "./slices/esicSlice";
import profileSlice from "./slices/profileSlice";
import navigationSlice from "./slices/navigationSlice";

export const store = configureStore({
  reducer: {
    auth: authSlice,
    aadhaar: aadhaarSlice,
    pan: panSlice,
    bank: bankSlice,
    profile: profileSlice,
    esic: esicSlice,
    navigation: navigationSlice
  },
});
