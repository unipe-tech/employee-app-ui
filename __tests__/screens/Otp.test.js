import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";
import { Provider } from "react-redux";
import OTPScreen from "../../screens/00_login/OTPScreen";
import { store } from "../../store/store";
import { shallow } from "enzyme";

let navigation;
beforeEach(() => {
  navigation = {
    dispatch: jest.fn(),
    navigate: jest.fn(),
  };
});

describe("OTP Screen", () => {
  // snapshot test
  it("should render OTPScreen UI properly", () => {
    const tree = shallow(
      <Provider store={store}>
        <NavigationContainer>
          <OTPScreen />
        </NavigationContainer>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  }, 8000);
});
