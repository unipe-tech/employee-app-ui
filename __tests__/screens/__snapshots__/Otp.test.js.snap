// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OTP Screen should render OTPScreen UI properly 1`] = `
<ContextProvider
  value={
    Object {
      "getServerState": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ForwardRef(NavigationContainerInner)>
    <OTPScreen />
  </ForwardRef(NavigationContainerInner)>
</ContextProvider>
`;
